scan.go

```diff
@@ -97,6 +98,26 @@ func (s *ScanService) SearchSource(searchSource *SearchSource) *ScanService {
 	return s
 }
 
+// Source allows the user to set the request body manually without using
+// any of the structs and interfaces in Elastic.
+func (s *ScanService) Source(sourceToSet interface{}) *ScanService {
+	switch sourceToSet.(type) {
```

Comment:
You shouldn't to the serialization part here. This is done later when the request is sent via `PerformRequest`: See [`Request.SetBody(...)`](https://github.com/olivere/elastic/blob/release-branch.v3/request.go#L38). Simply save `source` and use it in `Do(...)` if it's non-nil.